### PR TITLE
84 create a main function api document

### LINKED ISSUE #84: Create a main function API Document
## Feature Title

Create a main function API Document

## User Stories

Create an API document

## Acceptance Criteria

- [ ] Create a document
- [ ] Create ci pipeline

## Technical Details

- API Changes: None
- Database Changes: None
- Dependencies: None

## Additional Context

None

### PR DESCRIPTION
**Pull request by hitohata**
## Overviews of implementation

Added an OpenAPI document in the main project.
Changed the typedoc directory.

## Review points

Run docker-compose command

### CODE REVIEW

**Review by taniguchiiqqq — CHANGES_REQUESTED** [review]
Thank you! I ensured that the web-api doc works. Can you just add a docker compose command to readme?

I found the following errors happened when I run the container after I did once like `docker compose up`. Can you check if it happens in your environment? ( and sorry I didn't noticed in the creating type doc container PR. )
> nishiki-backend-project-main-type-doc-1      | mv: inter-device move failed: './docs/interfaces' to '/build/interfaces'; unable to remove target: Directory not empty

File: backend/main/web-api/main-web-api.yaml [hunk 1]
```diff
@@ -0,0 +1,441 @@
...
+                  type: string
+      responses:
+        "202":
+          description: update user's information
+        "401":
+          description: try to change other user's information
+    delete:
+      description: delete a user
+      tags:
+        - user
```
Comment (taniguchiiqqq) [inline backend/main/web-api/main-web-api.yaml:91]: I found a Semantic Error here. It needs to define the declared path parameter "userId" in this delete operation.
```yml
      parameters:
        - name: userId
          in: path
          description: ID of user
          required: true
          schema:
            type: string
```

**Review by taniguchiiqqq — CHANGES_REQUESTED** [review]

**Comment by hitohata** [comment]
@taniguchiiqqq 
>I found the following errors happened when I run the container after I did once like docker compose up. Can you check if it happens in your environment? ( and sorry I didn't noticed in the creating type doc container PR. )

Thank you for reviewing.
It seems that it comes from the target directory is not empty. This is no problem when deploying because the target directory which is in the docker memory is initialized every time. But, yes, it's not comfortable for us. So, I'll manage it somehow.

**Comment by taniguchiiqqq** [comment]
@hitohata 
ohh sorry, I forgot it's mainly for the GithubPages. I'll approve just after you fix the semantic error.

**Comment by hitohata** [comment]
@taniguchiiqqq 
OK, I fixed it.

**Review by taniguchiiqqq — APPROVED** [review]

**Pull request merged**